satellite/satellitedb/migrate_postgres_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+
+	"storj.io/storj/internal/dbutil/dbschema"
+	"storj.io/storj/internal/dbutil/pgutil"
+	"storj.io/storj/satellite/satellitedb"
+	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+)
+
+// loadSnapshots loads all the dbschemas from testdata/postgres.* caching the result
+func loadSnapshots(connstr string) (*dbschema.Snapshots, error) {
+	snapshots := &dbschema.Snapshots{}
+
+	// find all postgres sql files
+	matches, err := filepath.Glob("testdata/postgres.*")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, match := range matches {
+		versionStr := match[19 : len(match)-4] // hack to avoid trim issues with path differences in windows/linux
+		version, err := strconv.Atoi(versionStr)
+		if err != nil {
+			return nil, err
+		}
+
+		scriptData, err := ioutil.ReadFile(match)
+		if err != nil {
+			return nil, err
+		}
+
+		snapshot, err := pgutil.LoadSnapshotFromSQL(connstr, string(scriptData))
+		if err != nil {
+			return nil, err
+		}
+		snapshot.Version = version
+
+		snapshots.Add(snapshot)
+	}
+
+	return snapshots, nil
+}
+
+const newDataSeparator = `-- NEW DATA --`
+
+func newData(snap *dbschema.Snapshot) string {
+	tokens := strings.SplitN(snap.Script, newDataSeparator, 2)
+	if len(tokens) != 2 {
+		return ""
+	}
+	return tokens[1]
+}
+
+var (
+	dbxschema struct {
+		sync.Once
+		*dbschema.Schema
+		err error
+	}
+)
+
+// loadDBXSChema loads dbxscript schema only once and caches it,
+// it shouldn't change during the test
+func loadDBXSchema(connstr, dbxscript string) (*dbschema.Schema, error) {
+	dbxschema.Do(func() {
+		dbxschema.Schema, dbxschema.err = pgutil.LoadSchemaFromSQL(connstr, dbxscript)
+	})
+	return dbxschema.Schema, dbxschema.err
+}
+
+func TestMigratePostgres(t *testing.T) {
+	if *satellitedbtest.TestPostgres == "" {
+		t.Skip("Postgres flag missing, example: -postgres-test-db=" + satellitedbtest.DefaultPostgresConn)
+	}
+
+	snapshots, err := loadSnapshots(*satellitedbtest.TestPostgres)
+	require.NoError(t, err)
+
+	for _, base := range snapshots.List {
+		// versions 0 to 4 can be a starting point
+		if base.Version < 0 || 4 < base.Version {
```

Comment:
Maybe a constant for this? As far as I understand it will increase over time.

Reply:
No, it shouldn't... actually once we upgrade, having a single snapshot should be sufficient.

Reply:
note I did add a constant.

but, the reason we don't need more base versions is because we either:
1. start from clean database
2. start from a database where we know the exact version

Currently the database can be in any of those states from 0 to 4.